tests/components/purpleair/test_config_flow.py

```diff
@@ -245,6 +326,90 @@ async def test_options_add_sensor_duplicate(hass, config_entry, setup_purpleair)
     assert result["reason"] == "already_configured"
 
 
+@pytest.mark.parametrize(
+    "get_sensor_mock,get_sensor_errors",
+    [
+        (AsyncMock(side_effect=Exception), {"base": "unknown"}),
+        (AsyncMock(side_effect=PurpleAirError), {"base": "unknown"}),
+        (AsyncMock(side_effect=NotFoundError), {"base": "unknown_sensor_index"}),
+    ],
+)
+async def test_options_add_sensor_by_sensor_index(
+    hass, api, config_entry, get_sensor_errors, get_sensor_mock, setup_purpleair
+):
+    """Test adding a sensor by sensor index via the options flow (including errors)."""
+    result = await hass.config_entries.options.async_init(config_entry.entry_id)
+    assert result["type"] == data_entry_flow.FlowResultType.MENU
+    assert result["step_id"] == "init"
+
+    result = await hass.config_entries.options.async_configure(
+        result["flow_id"], user_input={"next_step_id": "choose_add_sensor_method"}
+    )
+    assert result["type"] == data_entry_flow.FlowResultType.MENU
+    assert result["step_id"] == "choose_add_sensor_method"
+
+    result = await hass.config_entries.options.async_configure(
+        result["flow_id"], user_input={"next_step_id": "add_sensor_by_index"}
+    )
+    assert result["type"] == data_entry_flow.FlowResultType.FORM
+    assert result["step_id"] == "add_sensor_by_index"
+
+    # Test errors that can arise when searching for nearby sensors:
+    with patch.object(api.sensors, "async_get_sensor", get_sensor_mock):
+        result = await hass.config_entries.options.async_configure(
+            result["flow_id"],
+            user_input={
+                "sensor_index": "567890",
+            },
+        )
+        assert result["type"] == data_entry_flow.FlowResultType.FORM
+        assert result["errors"] == get_sensor_errors
+
+    result = await hass.config_entries.options.async_configure(
+        result["flow_id"],
+        user_input={
+            "sensor_index": "567890",
+        },
+    )
+    assert result["type"] == data_entry_flow.FlowResultType.CREATE_ENTRY
+    assert result["data"] == {
+        "last_update_sensor_add": True,
+        "sensor_indices": [123456, 567890],
```

Comment:
Minor suggestion: if you define these indices as `TEST_SENSOR1_INDEX` and `TEST_SENSOR2_INDEX` that might add nice context throughout the rest of this test module where they're used.

Reply:
Thank you! I actually have this covered here: https://github.com/home-assistant/core/pull/84894.